mne/viz/backends/tests/test_renderer.py

```diff
@@ -217,16 +217,20 @@ def fail(x):
 def test_3d_warning(renderer_pyvistaqt, monkeypatch):
     """Test that warnings are emitted for old Mesa."""
     fig = renderer_pyvistaqt.create_3d_figure((800, 600))
-    _is_mesa = renderer_pyvistaqt.backend._is_mesa
+    from mne.viz.backends._pyvista import _is_osmesa
+
     plotter = fig.plotter
     good = "OpenGL renderer string: OpenGL 3.3 (Core Profile) Mesa 20.0.8 via llvmpipe (LLVM 10.0.0, 256 bits)\n"  # noqa
     bad = "OpenGL renderer string: OpenGL 3.3 (Core Profile) Mesa 18.3.4 via llvmpipe (LLVM 7.0, 256 bits)\n"  # noqa
     monkeypatch.setattr(platform, "system", lambda: "Linux")  # avoid short-circuit
     monkeypatch.setattr(plotter.ren_win, "ReportCapabilities", lambda: good)
-    assert _is_mesa(plotter)
+    assert _is_osmesa(plotter)
     monkeypatch.setattr(plotter.ren_win, "ReportCapabilities", lambda: bad)
     with pytest.warns(RuntimeWarning, match=r"18\.3\.4 is too old"):
-        assert _is_mesa(plotter)
+        assert _is_osmesa(plotter)
     non = "OpenGL 4.1 Metal - 76.3 via Apple M1 Pro\n"
     monkeypatch.setattr(plotter.ren_win, "ReportCapabilities", lambda: non)
-    assert not _is_mesa(plotter)
+    assert not _is_osmesa(plotter)
+    non = "OpenGL renderer string: llvmpipe\n"
+    monkeypatch.setattr(plotter.ren_win, "ReportCapabilities", lambda: non)
+    assert not _is_osmesa(plotter)
```

Comment:
here you assert that plotter is *not* OSMESA if its renderer string is "llvmpipe" but above I see `is_osmesa = renderer[0] == "llvmpipe"` which seems to say the opposite.  Am I misunderstanding?

Reply:
Hah, this is due to a spacing error. Will fix...